text/0497-glue-l2-construct.md

```diff
@@ -0,0 +1,418 @@
+  scriptS3Url: 's3://bucket-name/path-to-python-script',
+  minWorkers: 20,
+  numberOfWorkers: 50
+});
+```
+
+### Job Triggers
+
+We will add convenience functions for adding triggers to jobs. Standalone
+triggers are an anti-pattern, so we will only create triggers from within a
+workflow.
+
+1. **On Demand Triggers**
+
+On demand triggers can start glue jobs or crawlers. We’ll add convenience
+functions to create on-demand crawler or job triggers. The trigger method
+will take an optional description but abstract the requirement of an actions
+list using the job or crawler name.
+
+```
+myGlueJob.createOnDemandTrigger(this, 'MyJobTrigger', {
+  description: 'On demand run for ' + myGlueJob.name,
+});
+```
+
+```
+myGlueCrawler.createOnDemandTrigger(this, 'MyCrawlerTrigger');
+```
+
+1. **Scheduled Triggers**
+
+Schedule triggers are a way for customers to create jobs using cron
+expressions. We’ll provide daily, weekly and hourly options which customer
+can override using custom cron expression. The trigger method will take an
+optional description but abstract the requirement of an actions list using
+the job or crawler name. \
+
+```
+myGlueJob.createDailyTrigger(this, 'MyDailyTrigger');
+
+myGlueJob.createHourlyTrigger(this, 'MyHourlyTrigger');
+
+myGlueJob.createWeeklyTrigger(this, 'MyWeeklyTrigger');
+
+myGlueJob.createScheduledTrigger(this, 'MyScheduledTrigger', {
+  description: 'Scheduled run for ' + crawler.name,
+  schedule: '`cron(15 12 * * ? *)'`` //``every day at 12:15 UTC`
+});
+```
+
+#### **3. Notify  Event Trigger**
+
+This type of trigger is only supported with Glue workflow. There are two types
+of notify event triggers, batching and non-batching trigger. For batching trigger
+`BatchSize` customer has to specify but for non-batching `BatchSize` will be set
+to 1. For both trigger type `BatchWindow will be default to 900 seconds`
+
+```
+myGlueJob.createNotifyEventBatchingTrigger(this, 'MyNotifyTrigger', batchSize,
+    workFlowName: workflow.name,
+    batchSize: batchSize
+);
+
+myGlueCrawler.createNotifyEventBatchingTrigger(this, 'MyNotifyTrigger', batchSize,
+    workFlowName: workflow.name,
+    batchSize: batchSize
+);
+
+myGlueJob.createNotifyEventNonBatchingTrigger(this, 'MyNotifyTrigger',
+    workFlowName: workflow.name
+);
+
+myGlueCrawler.createNotifyEventNonBatchingTrigger(this, 'MyNotifyTrigger',
+    workFlowName: workflow.name
+);
+
+```
+
+#### **4. Conditional Trigger**
+
+Conditional trigger has predicate and action associated with it. Based on
+predicate, trigger action will be executed.
+
+```
+// Triggers on Job and Crawler status
+myGlueJob.addConditionalTrigger(
+  jobs: [
+    {jobArn: "job1-arn", status: glue.JobStatus.SUCCEEDED},
+    {jobArn: "job2-arn", status: glue.JobStatus.FAILED},
+  ], crawlers: [
+    {crawlerArn: "crawler1-arn", status: glue.CrawlerStatus.SUCCEEDED},
+    {crawlerArn: "crawler2-arn", status: glue.CrawlerStatus.TIMEOUT},
+]);
+
+```
+
+### Connection Properties
```

Comment:
Could you provide examples of what this contract would look like?

Reply:
Janny will work on codifying this for clarity.